python/core/geometry/qgsabstractgeometry.sip

```diff
@@ -1,406 +1,537 @@
+    if (dynamic_cast<QgsPointV2*>(sipCpp) != NULL)
+           sipType = sipType_QgsPointV2;
+         else if (dynamic_cast<QgsLineString*>(sipCpp) != NULL )
+           sipType = sipType_QgsLineString;
+         else if (dynamic_cast<QgsCircularString*>(sipCpp) != NULL )
+           sipType = sipType_QgsCircularString;
+         else if (dynamic_cast<QgsCompoundCurve*>(sipCpp) != NULL )
+           sipType = sipType_QgsCompoundCurve;
+         else if (dynamic_cast<QgsTriangle*>(sipCpp) != NULL )
+           sipType = sipType_QgsTriangle;
+         else if (dynamic_cast<QgsPolygonV2*>(sipCpp) != NULL )
+           sipType = sipType_QgsPolygonV2;
+         else if (dynamic_cast<QgsCurvePolygon*>(sipCpp) != NULL )
+           sipType = sipType_QgsCurvePolygon;
+         else if (dynamic_cast<QgsMultiPointV2*>(sipCpp) != NULL )
+           sipType = sipType_QgsMultiPointV2;
+         else if (dynamic_cast<QgsMultiLineString*>(sipCpp) != NULL )
+           sipType = sipType_QgsMultiLineString;
+         else if (dynamic_cast<QgsMultiPolygonV2*>(sipCpp) != NULL )
+           sipType = sipType_QgsMultiPolygonV2;
+         else if (dynamic_cast<QgsMultiSurface*>(sipCpp) != NULL )
+           sipType = sipType_QgsMultiSurface;
+         else if (dynamic_cast<QgsMultiCurve*>(sipCpp) != NULL )
+           sipType = sipType_QgsMultiCurve;
+         else if (dynamic_cast<QgsGeometryCollection*>(sipCpp) != NULL )
+           sipType = sipType_QgsGeometryCollection;
+         else
+           sipType = 0;
+%End
+  public:
+
     enum SegmentationToleranceType
     {
-        MaximumAngle = 0,
-        MaximumDifference
+      MaximumAngle,
+      MaximumDifference
     };
 
-
     QgsAbstractGeometry();
     virtual ~QgsAbstractGeometry();
-    QgsAbstractGeometry( const QgsAbstractGeometry& geom );
-    //virtual QgsAbstractGeometry& operator=( const QgsAbstractGeometry& geom );
+    QgsAbstractGeometry( const QgsAbstractGeometry &geom );
 
-    /** Clones the geometry by performing a deep copy
-     */
-    virtual QgsAbstractGeometry* clone() const = 0 /Factory/;
+    virtual QgsAbstractGeometry *clone() const = 0 /Factory/;
+%Docstring
+ Clones the geometry by performing a deep copy
+ :rtype: QgsAbstractGeometry
+%End
 
-    /** Clears the geometry, ie reset it to a null geometry
-     */
     virtual void clear() = 0;
+%Docstring
+ Clears the geometry, ie reset it to a null geometry
+%End
 
-    /** Returns the minimal bounding box for the geometry
-     */
     virtual QgsRectangle boundingBox() const = 0;
+%Docstring
+ Returns the minimal bounding box for the geometry
+ :rtype: QgsRectangle
+%End
+
 
-    //mm-sql interface
-    /** Returns the inherent dimension of the geometry. For example, this is 0 for a point geometry,
-     * 1 for a linestring and 2 for a polygon.
-     */
     virtual int dimension() const = 0;
-    //virtual int coordDim() const { return mCoordDimension; }
```

Comment:
this is commented in the source too
https://github.com/qgis/QGIS/blob/7c79803cc37bd2bd79ca5c0c2d7f61b46eab61bc/python/core/geometry/qgsabstractgeometry.sip#L105